src/macrobrowserui.h

```diff
@@ -23,17 +23,24 @@ class MacroBrowserUI : public QDialog
 private slots:
     void onRequestError();
     void onRequestCompleted();
-    void requestMacroList(const QString &path="",const bool &directURL=false);
-    void itemClicked(QTableWidgetItem *item);
+    void requestMacroList(QTreeWidgetItem *item=nullptr,const bool &isFile=false);
+    void slotItemExpanded(QTreeWidgetItem *item);
+    void slotCurrentItemChanged(QTreeWidgetItem *item);
+    void slotItemClicked(QTreeWidgetItem *item);
+
+private:
+    static const int FileRole;
+    static const int UrlRole;
+    static const int PathRole;
+    static const int PopulatedRole;
 
 protected:
-    QTableWidget *tableWidget;
+    QTreeWidget *treeWidget;
     QDialogButtonBox *buttonBox;
     QLineEdit *leName;
     QPlainTextEdit *teDescription;
-    QString currentPath;
     QHash<QString,QString> cache;
-    QHash<QString,QList<QTableWidgetItem *> > itemCache;
+    QMap<QString,QList<QTreeWidgetItem *> > itemCache;
```

Comment:
the itemcache is not necessary because you place the item in the tree widget where they stay until destroyed.
For the table, a cache was necessary as items become invisible (and deleted from the table) when leaving a folder.

Reply:
still in header